# Importing the libraries

In [1]:
import numpy as np
import pandas as pd
from scipy import stats

# Loading the data

In [2]:
cust_data = pd.read_csv('Costomer+OrderForm.csv')

In [3]:
cust_data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [5]:
# we will use chi2_contigency because we don't know the distribution and we have to test whether the groups is having same distribution or not
stats.chi2_contingency([cust_data['Phillippines'].value_counts(),cust_data['Indonesia'].value_counts(),cust_data['Malta'].value_counts(),cust_data['India'].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

### chi2 = 3.8589606858203545
### p-value = 0.2771020991233144
### dof = 3
### expected - ndarray = array([[271.75,  28.25],
###                                                 [271.75,  28.25],
###                                                 [271.75,  28.25],
###                                                 [271.75,  28.25]])

## We can see p-value > alpha(0.05)
## Therefore, H0 is accepted